src/open_company_web/components/topic_list.cljs

```diff
@@ -179,7 +177,133 @@
+        col2-pinned-topics (sel [:div.col-2 :div.topic.draggable-topic])
+        col2-pinned-topics-list (vec (for [topic col2-pinned-topics] (.data (js/$ topic) "section")))
+        col3-pinned-topics (sel [:div.col-3 :div.topic.draggable-topic])
+        col3-pinned-topics-list (vec (for [topic col3-pinned-topics] (.data (js/$ topic) "section")))
+        max-count (max (count col1-pinned-topics-list) (count col2-pinned-topics-list) (count col3-pinned-topics-list))
+        pinned-topics-list (loop [topics []
+                                  idx 0]
+                             (if (<= (inc idx) max-count)
+                               (recur (vec (remove nil? (conj topics (get col1-pinned-topics-list idx) (get col2-pinned-topics-list idx) (get col3-pinned-topics-list idx))))
+                                      (inc idx))
+                               topics))
+        other-topics (sel [:div.topic.not-draggable-topic])
+        other-topics-list (vec (for [topic other-topics] (.data (js/$ topic) "section")))]
+    (dispatcher/dispatch! [:new-sections (vec (concat pinned-topics-list other-topics-list))])))
+
+(defn pinned-count [data]
+  (let [company-data (:company-data data)
+        {:keys [pinned]} (utils/get-pinned-other-keys (:sections company-data) company-data)]
+    (count pinned)))
+
+(defn coord-inside [left top topic]
+  (if topic
+    (let [dragging-topic (.data (js/$ ".ui-draggable-dragging") "topic")
+          topic-el (js/$ (str ".topic-row[data-topic=" (name topic) "]"))
+          topic-pos (.position topic-el)
+          topic-position (utils/absolute-offset (.get topic-el 0))
+          topic-posalter {:left (- (gobj/get topic-pos "left") 20)
+                          :top (- (gobj/get topic-pos "top") 30)}
+          topic-size {:width (.width topic-el) :height (.height topic-el)}
+          target-css {:width (int (+ (:width topic-size) 26))
+                      :height (int (+ (:height topic-size) 22))
+                      :left (int (:left topic-position))
+                      :top (int (- (:top topic-position) 84))}]
+      (if (and (not= (name topic) dragging-topic)
+                 (>= left (:left target-css) )
+                 (>= top (:top target-css))
+                 (< left (+ (:left target-css) (:width target-css)))
+                 (< top (+ (:top target-css) (:height target-css))))
+        (if (< (- left (:left topic-position)) (/ (:width topic-size) 2))
+          {:side "left"
+           :topic-el topic-el
+           :inside? true
+           :topic topic}
+          {:side "right"
+           :topic-el topic-el
+           :inside? true
+           :topic topic})
+        {:topic-el topic-el
+         :inside? false}))
+    (sentry/capture-message (str "open-company-web.components.topic-list/coord-inside params, left:" left ", top:" top ", topic:" topic))))
+
+(defn get-topic-at-position [owner left top stop?]
```

Comment:
function docstring and possibly some inline comments would be good here. this function is non-trivial.